CADETProcess/optimization/surrogate.py

```diff
@@ -0,0 +1,381 @@
+
+        self.surrogate_model_F: BaseEstimator = None
+        self.surrogate_model_G: BaseEstimator = None
+        self.surrogate_model_M: BaseEstimator = None
+        self.surrogate_model_CV: BaseEstimator = None
+
+        self.fit_gaussian_process()
+
+
+    def uncondition(self):
+        """
+        Reset the optimization problem bounds to their original values.
+        """
+        for var, lb, ub in zip(
+            self.optimization_problem.variables,
+            self.lower_bounds_copy,
+            self.upper_bounds_copy
+        ):
+            var.lb = lb
+            var.ub = ub
+
+    def fit_gaussian_process(self):
+        """
+        Fit Gaussian process surrogate models to the population.
+
+        Parameters:
+        - population (np.ndarray): The population for fitting the surrogate models.
+        """
+        X = self.population.x_untransformed
+        F = self.population.f
+        G = self.population.g
+        M = self.population.m
+        CV = self.population.cv
+
+        gp_f = GaussianProcessRegressor()
+        gp_f.fit(X, F)
+        self.surrogate_model_F = gp_f
+
+        if G is not None:
+            gp_g = GaussianProcessRegressor()
+            gp_g.fit(X, G)
+            self.surrogate_model_G = gp_g
+
+        if M is not None:
+            gp_m = GaussianProcessRegressor()
+            gp_m.fit(X, M)
+            self.surrogate_model_M = gp_m
+
+        if CV is not None:
+            gp_cv = GaussianProcessClassifier()
+            gp_cv.fit(X, CV)
+            self.surrogate_model_CV = gp_cv
+
+    def estimate_objectives(self, X, return_std=False):
+        """
+        Estimate the objectives using the surrogate model.
+
+        Parameters:
+        - X (np.ndarray): The input samples.
+        - return_std (bool, optional): Whether to return the standard deviation
+        of the predictions. Defaults to False.
+
+        Returns:
+        - np.ndarray: The estimated objectives.
+        """
+        return self.surrogate_model_F.predict(X, return_std=return_std)
+
+
+    def estimate_non_linear_constraints(self, X):
+        """
+        Estimate the non-linear constraints using the surrogate model.
+
+        Parameters:
+        - X (np.ndarray): The input samples.
+
+        Returns:
+        - Tuple[np.ndarray, np.ndarray]: The estimated non-linear
+        constraints (G, CV).
+        """
+        G_est = self.surrogate_model_G.predict(X)
+        CV_est = self.surrogate_model_CV.predict(X)
+        return G_est, CV_est
+
+    def estimate_meta_scores(self, X):
+        """
+        Estimate the meta scores using the surrogate model.
+
+        Parameters:
+        - X (np.ndarray): The input samples.
+
+        Returns:
+        - np.ndarray: The estimated meta scores.
+        """
+        M_est = self.surrogate_model_M.predict(X)
+        return M_est
+
+    def get_conditional_and_free_indices(self, conditional_vars={}):
```

Comment:
Using mutable objects as default arguments is dangerous.

Better use 

```
def get_conditional_and_free_indices(self, conditional_vars=None):
    if conditional_vars = None:
        conditional_vars  = {}
```

Otherwise, all function calls will potentially refer to the same dict object.